corehq/apps/userreports/util.py

```diff
@@ -331,3 +331,34 @@ def wrap_report_config_by_type(config):
         }[config["doc_type"]].wrap(config)
     except KeyError:
         raise ReportConfigurationNotFoundError()
+
+
+def find_in_json(json_obj, lookup_key, vals):
```

Comment:
You could use `jsonpath` to do this:
```
from jsonpath_ng.ext import parser
parser.parse("$..*[?name='asprin']").find(data)

Reply:
I tried using `jsonpath` but it did not work out the way it is expected.
see https://github.com/dimagi/commcare-hq/pull/31161/commits/c129d64be9cc290451b526b7c5b07d6f430b428e despite the object having the key and value in the object the `jsonpath` did not return anything. 
I have code commented out in the commit. 
Not sure if I am using it correctly.

Reply:
Ah, to match on children of a list you need to use this `$..[*][?type='related_doc']`

Reply:
It worked like a charm ⭐ 
Thanks for your help, Simon 😄